### PR TITLE
[zh-cn]: Update translation of Promise.constructor

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#13733

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 1]
```diff
@@ -99,6 +210,7 @@ function myAsyncFunction(url) {
 
 {{Compat}}
 
-## See also
+## 参见
 
-- [Using Promises](/zh-CN/docs/Web/JavaScript/Guide/Using_promises)
+- [`core-js` 中 `Promise` 的 polyfill](https://github.com/zloirock/core-js#ecmascript-promise)
+- [使用 Promises](/zh-CN/docs/Web/JavaScript/Guide/Using_promises)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:216]:
```suggestion
- [使用 Promise](/zh-CN/docs/Web/JavaScript/Guide/Using_promises)
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 2]
```diff
@@ -5,92 +5,203 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/Promise
...
+  // 任何依赖于返回结果的操作都必须在这个回调函数内定义。
+});
+// `readFile` 请求被发出后，此处的代码会立即执行。
+// 它不会等待回调函数被调用，因此使 `readFile` 成为了“异步”的。
+```
+
+为了利用 Promise 提供的更好的可读性和语言特性，`Promise()` 构造函数允许将基于回调的 API 转换为基于 Promise 的 API。
 
-    ```
+> **备注：** 如果你的任务已经基于 Promise 实现，你可能不需要使用 `Promise()` 构造函数。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:44]:
```suggestion
> **备注：** 如果你的任务已经基于 Promise 实现，你大概率不需要使用 `Promise()` 构造函数。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 3]
```diff
@@ -5,92 +5,203 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/Promise
...
+`executor` 的完成状态对 Promise 的状态影响有限：
 
-    1. `executor` 内的操作是异步的，并且提供一个回调（callback）。
-    2. 该回调在 `executor` 内定义。
-    3. 该回调通过调用 `resolutionFunc` 终止。
-    4. `resolutionFunc` 的调用包含一个 `value` 参数。
-    5. 该`value` 被返回给绑定的 `Promise` 对象上。
-    6. 该`Promise` 对象（异步地）调用任何相关的 `.then(handleResolved)`。
-    7. `.then(handleResolved)` 收到的 `value` 作为入参被传递给了 `handleResolved` 的调用。（参见[Promise 的链式调用](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#promise%E7%9A%84%E9%93%BE%E5%BC%8F%E8%B0%83%E7%94%A8)）
+- `executor` 函数的返回值会被忽略。`executor` 函数中的 `return` 语句仅影响控制流程，改变函数是否执行的部分，但不会影响 Promise 的履行值。如果 `executor` 函数退出，且未来不可能调用 `resolveFunc` 或 `rejectFunc`（例如，没有安排异步任务），那么 Promise 将永远保持待定状态。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:65]:
```suggestion
- `executor` 函数的返回值会被忽略。`executor` 函数中的 `return` 语句仅影响控制流程，控制函数某个部分是否执行，但不会影响 Promise 的履行值。如果 `executor` 函数退出，且未来不可能调用 `resolveFunc` 或 `rejectFunc`（例如，没有安排异步任务），那么 Promise 将永远保持待定状态。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 4]
```diff
@@ -5,92 +5,203 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/Promise
...
 
-## 示例
+1. 在构造函数生成新的 `Promise` 对象时，它还会生成一对相应的 `resolveFunc` 和 `rejectFunc` 函数；它们与 `Promise` 对象“绑定”在一起。
+2. `executor` 通常会封装某些提供基于回调的 API 的异步操作。回调函数（传给原始回调 API 的函数）在 `executor` 代码中定义，因此它可以访问 `resolveFunc` 和 `rejectFunc`。
+3. `executor` 是同步调用的（在构造 `Promise` 时立即调用），并将 `resolveFunc` 和 `rejectFunc` 函数作为传入参数。
+4. `executor` 中的代码有机会执行某些操作。异步任务的最终完成通过 `resolveFunc` 或 `rejectFunc` 引起的副作用与 Promise 实例进行通信。这个副作用让 `Promise` 对象变为“已解决”状态。
+   - 如果先调用 `resolveFunc`，则传入的值将[解决](#resolver_函数)。Promise 可能会保持处于待定状态（如果传入了另一个 [thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 对象），变为已兑现状态（在传入非 thenable 值的大多数情况下），或者变为已拒绝状态（在无效的解析值的情况下）。
+   - 如果先调用 `rejectFunc`，则 Promise 立即变为已拒绝状态。
+   - 一旦 `resolveFunc` 或 `rejectFunc` 中的一个被调用，Promise 将保持解决状态。只有第一次调用 `resolveFunc` 或 `rejectFunc` 会影响 Promise 的最终状态，随后对任一函数的调用都不能更改兑现值或拒绝原因，也不能将其最终状态从“已兑现”转换为“已拒绝”或相反。
+   - 如果 `executor` 抛出错误，则 Promise 被拒绝。但是，如果 resolveFunc 或 rejectFunc 中的一个被调用（因此 Promise 已经被解决），则忽略该错误。
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:79]:
```suggestion
   - 如果 `executor` 抛出错误，则 Promise 被拒绝。但是，如果 resolveFunc 或 rejectFunc 中的一个已经被调用（因此 Promise 已经被解决），则忽略该错误。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 5]
```diff
@@ -5,92 +5,203 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/Promise
...
+      (value) => resolve(value),
+      (reason) => reject(reason),
+    );
+  } catch (e) {
+    reject(e);
+  }
+});
+```
+
+在 `resolve(thenable)` 的情况下：
```
Comment (Josh-Cena) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:133]:
```suggestion
除了在 `resolve(thenable)` 的情况中，有如下区别：
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 6]
```diff
@@ -5,92 +5,203 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/Promise
...
-    3. 该回调通过调用 `resolutionFunc` 终止。
-    4. `resolutionFunc` 的调用包含一个 `value` 参数。
-    5. 该`value` 被返回给绑定的 `Promise` 对象上。
-    6. 该`Promise` 对象（异步地）调用任何相关的 `.then(handleResolved)`。
-    7. `.then(handleResolved)` 收到的 `value` 作为入参被传递给了 `handleResolved` 的调用。（参见[Promise 的链式调用](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#promise%E7%9A%84%E9%93%BE%E5%BC%8F%E8%B0%83%E7%94%A8)）
+- `executor` 函数的返回值会被忽略。`executor` 函数中的 `return` 语句仅影响控制流程，调整函数某个部分是否执行，但不会影响 Promise 的履行值。如果 `executor` 函数退出，且未来不可能调用 `resolveFunc` 或 `rejectFunc`（例如，没有安排异步任务），那么 Promise 将永远保持待定状态。
+- 如果在 `executor` 函数中抛出错误，则 Promise 将被拒绝，除非 `resolveFunc` 或 `rejectFunc` 已经被调用。
 
-### 返回值
+> **备注：** 待定的 Promise 的存在并不会防止程序退出。如果事件循环为空，则程序会退出，尽管存在待定的 Promise（因为它们必然永远处于待定状态）。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:68]:
```suggestion
> **备注：** 待定的 Promise 的存在并不会阻止程序退出。如果事件循环为空，则程序会退出，尽管存在待定的 Promise（因为它们必然永远处于待定状态）。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 7]
```diff
@@ -5,92 +5,203 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/Promise
...
 
-当通过 `new` 调用时，`Promise` 构造函数返回一个 promise 对象。当 `resolutionFunc` 或者 `rejectionFunc` 被调用时，该 promise 对象将会“被敲定”。注意，如果您调用 `resolutionFunc` 或者 `rejectionFunc` 时将另一个 Promise 对象作为参数，您可以称其“被敲定（resolved）”，但仍不能称其“被解决（settled）”。
+以下是典型的 Promise 流程概述：
 
-## 示例
+1. 在构造函数生成新的 `Promise` 对象时，它还会生成一对相应的 `resolveFunc` 和 `rejectFunc` 函数；它们与 `Promise` 对象“绑定”在一起。
+2. `executor` 通常会封装某些提供基于回调的 API 的异步操作。回调函数（传给原始回调 API 的函数）在 `executor` 代码中定义，因此它可以访问 `resolveFunc` 和 `rejectFunc`。
+3. `executor` 是同步调用的（在构造 `Promise` 时立即调用），并将 `resolveFunc` 和 `rejectFunc` 函数作为传入参数。
+4. `executor` 中的代码有机会执行某些操作。异步任务的最终完成通过 `resolveFunc` 或 `rejectFunc` 引起的副作用与 Promise 实例进行通信。这个副作用让 `Promise` 对象变为“已解决”状态。
+   - 如果先调用 `resolveFunc`，则传入的值将[解决](#resolver_函数)。Promise 可能会保持处于待定状态（如果传入了另一个 [thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 对象），变为已兑现状态（在传入非 thenable 值的大多数情况下），或者变为已拒绝状态（在无效的解析值的情况下）。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:76]:
```suggestion
   - 如果先调用 `resolveFunc`，则传入的值将[解决](#resolver_函数)。Promise 可能会保持待定状态（如果传入了另一个 [thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 对象），变为已兑现状态（在传入非 thenable 值的大多数情况下），或者变为已拒绝状态（在解析值无效的情况下）。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 8]
```diff
@@ -5,92 +5,203 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/Promise
...
+  });
+
+readFilePromise("./data.txt")
+  .then((result) => console.log(result))
+  .catch((error) => console.error("读取文件失败"));
+```
+
+### Resolver 函数
+
+`resolveFunc` 解析函数有以下行为：
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:104]: This is not a "parser".

```suggestion
`resolveFunc` 解决函数有以下行为：
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 9]
```diff
@@ -5,92 +5,203 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/Promise
...
+  }
+});
+```
+
+但是在 `resolve(thenable)` 的情况中，有如下区别：
+
+1. `resolve` 函数是同步调用的，因此再次调用 `resolve` 或 `reject` 函数没有任何影响，即使通过 `anotherPromise.then()` 绑定的处理程序尚未被调用。
+2. `then` 方法是异步调用的，因此如果传入 thenable 对象，则该 `Promise` 对象不会被立即解决。
+
+因为 `resolve` 函数再次调用时使用 `thenable.then()` 传递给它的任何值作为 `value` 参数，所以解析函数能够展开嵌套的 thenable 对象，其中一个 thenable 对象调用其 `onFulfilled` 处理程序并返回另一个 thenable 对象。这样做的效果是，真实的 `Promise` 对象的兑现处理程序永远不会接收到 thenable 对象作为其兑现值。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:138]:
```suggestion
因为 `resolve` 函数再次调用时使用 `thenable.then()` 传递给它的任何值作为 `value` 参数，所以解决函数能够展开嵌套的 thenable 对象，其中一个 thenable 对象调用其 `onFulfilled` 处理程序并返回另一个 thenable 对象。这样做的效果是，真实的 `Promise` 对象的兑现处理器永远不会接收到 thenable 对象作为其兑现值。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md [hunk 10]
```diff
@@ -5,92 +5,203 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/Promise
...
+    new Promise((resolveInner) => {
+      setTimeout(() => {
+        resolveInner("内部");
+      }, 100);
+    }),
+  );
+});
+```
+
+`fulfilledResolved` Promise 对象在被解决时立即变为已兑现状态，因为它被解决为一个非 thenable 值。然而，在它被创建时，它是未解决的，因为 `resolve` 或 `reject` 函数还没有被调用。未解决的 Promise 对象必然是待定状态：
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/promise/index.md:176]:
```suggestion
`fulfilledResolved` Promise 对象在被解决时立即变为已兑现状态，因为它以非 thenable 值解决。然而，在它被创建时，它是未解决的，因为 `resolve` 或 `reject` 函数还没有被调用。未解决的 Promise 对象必然是待定状态：
```

**Review by yin1999 — COMMENTED** [review]
Could you also update the title of this document :)

**Comment by JasonLamv-t** [comment]
> Could you also update the title of this document :)

Already done.

**Review by yin1999 — APPROVED** [review]
Thanks @JasonLamv-t, @Josh-Cena :tada:

**Pull request merged**